src/lib/tls/tls12/tls_record.h

```diff
@@ -61,13 +63,11 @@ class Connection_Cipher_State final
 
       Nonce_Format nonce_format() const { return m_nonce_format; }
 
-      std::chrono::seconds age() const
-         {
-         return std::chrono::duration_cast<std::chrono::seconds>(
-            std::chrono::system_clock::now() - m_start_time);
-         }
+      std::chrono::seconds age() const;
```

Comment:
Or perhaps instead of having an `age` function at all, just expose `start_time` and in the code that needs the age, have it invoke the `Callback&` directly, eg `auto age = cb.tls_current_timestamp() - record.start_time()`

Reply:
In fact this method doesn't seem to be used at all anymore in the TLS 1.2 code base.